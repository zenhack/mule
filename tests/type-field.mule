#XFAIL: type member
{
  , type t = int
  , x : t = 4
}
