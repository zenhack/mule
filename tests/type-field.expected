
Desugared:

	(letrec (types (((t (rec t int))))) (values ((x (: (Integer 4) t))))
	 (.type= t (.= x x {}) t))

inferred type:

	(record (types ((t int))) (values ((x int))))

Runtime term:

	(LetRec
	 (((0 (Const (Integer 4))))
	  (Update (old (Record ())) (label x) (field (Var 0)))))

Evaluated:

	(Record ((x (Const (Integer 4)))))
